src/sdk/object_sdk.js

```diff
@@ -64,6 +65,40 @@ class ObjectSDK {
             this.bucketspace_fs = new BucketSpaceFS({ fs_root: process.env.BUCKETSPACE_FS });
         }
         this.requesting_account = undefined;
+        this.abort_controller = new AbortController();
+    }
+
+    /**
+     * setup_abort_controller adds event handlers to the http request and response,
+     * in order to handle aborting requests gracefully. The `abort_controller` member will
+     * be used to signal async flows that abort was detected.
+     * @see {@link https://nodejs.org/docs/latest/api/globals.html#class-abortcontroller}
+     * @param {import('http').IncomingMessage} req 
+     * @param {import('http').ServerResponse} res 
+     */
+    setup_abort_controller(req, res) {
+        res.once('error', err => {
+            dbg.log0('response error:', err, req.url);
+            this.abort_controller.abort();
+        });
+
+        req.once('aborted', () => {
+            dbg.log0('request aborted', req.url);
+            this.abort_controller.abort();
+        });
```

Comment:
Seems like we need to do this:
```
        req.once('close', () => {
            if (req.destroyed) {
                dbg.log0('request aborted', req.url);
                this.abort_controller.abort(new Error('request aborted ' + req.url));
            }
        });
```

Reply:
Ok, so this is a gap for moving to node 16, because on node 14 we can't check the req.destroyed state yet.

FYI @liranmauda and @nimrod-becker - we need to track this so that once we switch to node 16 we make sure to include this fix, which @romayalon left as a comment in the code.